packages/typespec-ts/src/modular/emitSamples.ts

```diff
@@ -0,0 +1,516 @@
+      : "";
+    const isPaging = method.kind === "paging";
+    const methodCall = `client.${prefix}${method.name}(${methodParams.join(
+      ", "
+    )})`;
+    if (isPaging) {
+      exampleFunctionBody.push(`const resArray = new Array();`);
+      exampleFunctionBody.push(
+        `for await (let item of ${methodCall}) { resArray.push(item); }`
+      );
+      exampleFunctionBody.push(`console.log(resArray);`);
+    } else {
+      exampleFunctionBody.push(`const result = await ${methodCall};`);
+      exampleFunctionBody.push(`console.log(result);`);
+    }
+
+    // Create a function declaration structure
+    const description = method.description ?? `execute ${method.name}`;
+    const normalizedDescription =
+      description.charAt(0).toLowerCase() + description.slice(1);
+    const functionDeclaration: FunctionDeclarationStructure = {
+      kind: StructureKind.Function,
+      isAsync: true,
+      name: exampleFunctionType.name,
+      statements: exampleFunctionType.body,
+      docs: [
+        `This sample demonstrates how to ${normalizedDescription}\n\n@summary ${normalizedDescription}\nx-ms-original-file: ${example.filePath}`
+      ]
+    };
+    sourceFile.addFunction(functionDeclaration);
+    exampleFunctions.push(exampleFunctionType.name);
+  }
+  // Add statements referencing the tracked declarations
+  const functions = exampleFunctions.map((f) => `${f}();`).join("\n");
+  sourceFile.addStatements(`
+  async function main() {
+    ${functions}
+  }
+
+  main().catch(console.error);`);
+  options.generatedFiles.push(sourceFile);
+  return sourceFile;
+}
+
+function buildParameterValueMap(example: SdkHttpOperationExample) {
+  const parameterMap: Record<string, SdkHttpParameterExample> = {};
+  example.parameters.forEach(
+    (param) =>
+      (parameterMap[
+        (param.parameter as any).serializedName ?? param.parameter.name
+      ] = param)
+  );
+  return parameterMap;
+}
+
+function prepareMethodExampleParameters(
+  dpgContext: SdkContext,
+  exampleName: string,
+  method: SdkServiceMethod<SdkServiceOperation>,
+  parameterMap: Record<string, SdkHttpParameterExample>
+): ExampleValue[] {
+  const parameters: ExampleValue[] = [];
+  for (const param of method.operation.parameters) {
+    if (
+      param.optional === true ||
+      param.onClient === true ||
+      param.type.kind === "constant"
+    ) {
+      continue;
+    }
+    const exampleValue = parameterMap[param.serializedName];
+    if (!exampleValue || !exampleValue.value) {
+      // report diagnostic if required parameter is missing
+      reportDiagnostic(dpgContext.program, {
+        code: "required-sample-parameter",
+        format: {
+          exampleName: exampleName,
+          paramName: param.name
+        },
+        target: NoTarget
+      });
+      continue;
+    }
+
+    parameters.push({
+      name: param.name,
+      value: getParameterValue(exampleValue.value),
+      isOptional: false
```

Comment:
why the parameter in method.operation.parameters is necessarily required??

Reply:
this parameters could be optional or required, but this part we mainly focus on the required parameters only. See [codes](https://github.com/marygao/autorest.typescript/blob/sample-gen-ready-branch/packages/typespec-ts/src/modular/emitSamples.ts#L260-L266).